application/src/main/java/org/thingsboard/server/config/CustomOAuth2AuthorizationRequestResolver.java

```diff
@@ -132,11 +139,18 @@ private OAuth2AuthorizationRequest resolve(HttpServletRequest request, String oa
             if (StringUtils.isEmpty(appToken)) {
                 throw new IllegalArgumentException("Invalid application token.");
             } else {
-                String appSecret = this.oAuth2ClientService.findAppSecret(new OAuth2ClientId(UUID.fromString(oauth2ClientId)), appPackage);
-                if (StringUtils.isEmpty(appSecret)) {
-                    throw new IllegalArgumentException("Invalid package: " + appPackage + ". No application secret found for Client Registration with given application package.");
+                String callbackUrlScheme;
+                if (platform != null) {
+                    callbackUrlScheme = validateMobileAppToken(oauth2ClientId, appPackage, PlatformType.valueOf(platform), appToken);
+                } else {
+                    // for backward compatibility with mobile apps of version 1.3.0 and less try to validate token with android and then ios app secret
+                    try {
+                        callbackUrlScheme = validateMobileAppToken(oauth2ClientId, appPackage, PlatformType.ANDROID, appToken);
+                    } catch (IllegalArgumentException e) {
+                        log.warn("Failed attempt to validate android application token", e);
```

Comment:
WARN is too much, let's use debug. Also, add some more info (oauth2ClientId, appPackage, appToken) to the log